pytorch_lightning/accelerators/gpu.py

```diff
@@ -48,6 +48,9 @@ def on_train_start(self) -> None:
             torch.cuda.empty_cache()
 
     def teardown(self) -> None:
+        # Note: this would be problematic for large model (which could not fit in one GPU)
+        # as FSDP module.to(device) would first summon all parameters
+        # (TODO: need to figure out solution)
```

Comment:
@shuyingsunshine21 as a short-term workaround, could we add `teardown` to the training type plugin? and have the accelerator's teardown call the training type plugin instead? 

this way we could start moving all the device-specific logic into each plugin to prep for the next refactor here

Reply:
sg, will have a separate PR for that.

Reply:
https://github.com/PyTorchLightning/pytorch-lightning/pull/7579 for reference.

Will remove this.

Reply:
IMO #7579 is not in line with the accelerator design and I believe we should not continue these patterns. I will comment there on the PR.